File: FinalProject(Backend)/src/main/java/com/finalproject/model/Activity.java
package com.finalproject.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "activity")
@AllArgsConstructor
@NoArgsConstructor
@Data
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Activity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "activityId")
    private Integer activityId;

    @Column(name = "activityName", nullable = false)
    private String activityName;

    @Column(name = "activityDetail", nullable = false, columnDefinition = "TEXT")
    private String activityDetail;

    @Column(name = "activityPrice", nullable = false)
    private Double activityPrice;

    @Column(name = "imagePaymentaActivity", nullable = false)
    private String imagePaymentaActivity;

    @Column(name = "activityDateTime", nullable = false)
    private Date activityDateTime;

    @JsonManagedReference
    @OneToMany(mappedBy = "activity", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<MemberTripActivity> memberTripActivity;

}

File: finalproject(Fontend)/lib/model/activity.dart
import 'package:finalproject/model/membertripactivity.dart';

class Activity {
  int? activityId;
  String? activityName;
  String? activityDetail;
  double? activityPrice;
  String? imagePaymentActivity;
  DateTime? activityDateTime;
  List<MemberTripActivity>? memberTripActivity;

  Activity({
    this.activityId,
    this.activityName,
    this.activityDetail,
    this.activityPrice,
    this.imagePaymentActivity,
    this.activityDateTime,
    this.memberTripActivity,
  });

  factory Activity.fromJson(Map<String, dynamic> json) => Activity(
    activityId: json['activityId'],
    activityName: json['activityName'],
    activityDetail: json['activityDetail'],
    activityPrice: (json['activityPrice'] as num?)?.toDouble(),
    imagePaymentActivity: (json['imagePaymentaActivity']), // ชื่อ field สะกดผิดจาก Java
    activityDateTime: json['activityDateTime'] != null
        ? DateTime.parse(json['activityDateTime'])
        : null,
    memberTripActivity: (json['memberTripActivity'] as List<dynamic>?)
        ?.map((e) => MemberTripActivity.fromJson(e))
        .toList(),
  );

  Map<String, dynamic> toJson() => {
    'activityId': activityId,
    'activityName': activityName,
    'activityDetail': activityDetail,
    'activityPrice': activityPrice,
    'imagePaymentaActivity': imagePaymentActivity, // ใช้ชื่อเดิมตาม backend
    'activityDateTime': activityDateTime?.toIso8601String(),
    'memberTripActivity': memberTripActivity?.map((e) => e.toJson()).toList(),
  };
}
